lsp-lens.el

```diff
@@ -375,9 +375,10 @@ CALLBACK - callback for the lenses."
   (interactive)
   (if (not lsp-lens-mode)
       (user-error "`lsp-lens-mode' not active")
+    (unless lsp-lens--overlays
+      (user-error "No lenses in current buffer"))
     (let* ((avy-action 'identity)
-           (action (cl-third
-                    (avy-process
+           (actions (avy-process
                      (-mapcat
```

Comment:
We can use `mapcan` here, since only calls to `-map-indexed`, which always returns a new list, are mapped.

Reply:
seems like it is fine to stay as it is. If we are looking for optimization here(which IMO is not the case since this is not a critical section) then we should drop at all the higher-order functions in favour of using `while`.

Reply:
It isn't about the optimization here; IMHO, using `mapcan` improves the code readability here, because it makes obvious that the returned lists of the inner lambda are newly-constructed, not taken from some global variable, because mutating them is Ok. This isn't the case for `dap-launch-configuration-providers`, for example, where we have to use `-mapcat`. Do you think I should leave it anyway?

Reply:
That is the whole benefit of using the pure function - you don't have to worry about the implementation and it will still work even if the inner code has changed. Using the non-pure function introduces one more thing to check when working with this code.

Reply:
Ok, given that this isn't performance critical, I'll leave this as-is.